sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2164,9 +2164,28 @@ case class Levenshtein(
     if (children.length > 3 || children.length < 2) {
       throw QueryCompilationErrors.wrongNumArgsError(
         toSQLId(prettyName), Seq(2, 3), children.length)
-    } else {
-      super.checkInputDataTypes()
     }
+    if (children.length == 3) {
+      threshold match {
+        case Some(e) if !e.foldable =>
```

Comment:
since the API takes `Column` now, I think it's fine to allow non-foldable threshold. We can do the check at runtime.

Reply:
Okay!